From 4991c494580904fe123cb1c4239764e16ae9d5ca Mon Sep 17 00:00:00 2001
From: Lenka Doudova <lryznaro@redhat.com>
Date: Thu, 23 Jan 2025 17:52:25 +0100
Subject: [PATCH] Issue 6568 - Fix failing webUI tests

Fixing webUI tests that are failing either due to changes or due to
existing bugs

Relates: https://github.com/389ds/389-ds-base/issues/6568

Author: Lenka Doudova

Reviewed by: Barbora Simonova
---
 dirsrvtests/tests/suites/webui/__init__.py    |  1 +
 .../suites/webui/database/database_test.py    | 73 ++++++++++++-------
 .../webui/ldap_browser/ldap_browser_test.py   |  1 +
 .../webui/monitoring/monitoring_test.py       |  3 +-
 4 files changed, 51 insertions(+), 27 deletions(-)

diff --git a/dirsrvtests/tests/suites/webui/__init__.py b/dirsrvtests/tests/suites/webui/__init__.py
index f92b75605..8333d0d49 100644
--- a/dirsrvtests/tests/suites/webui/__init__.py
+++ b/dirsrvtests/tests/suites/webui/__init__.py
@@ -194,6 +194,7 @@ def create_entry(frame, entry_type, entry_data):
         frame.get_by_role("button", name="Next", exact=True).click()
 
     elif entry_type == 'custom Entry':
+        frame.get_by_role("textbox", name="Search input").fill('account')
         frame.get_by_role("checkbox", name="Select row 0").check()
         frame.get_by_role("button", name="Next", exact=True).click()
         frame.get_by_role("checkbox", name="Select row 1").check()
diff --git a/dirsrvtests/tests/suites/webui/database/database_test.py b/dirsrvtests/tests/suites/webui/database/database_test.py
index d582bd185..ef105d262 100644
--- a/dirsrvtests/tests/suites/webui/database/database_test.py
+++ b/dirsrvtests/tests/suites/webui/database/database_test.py
@@ -52,7 +52,7 @@ def test_global_database_configuration_availability(topology_st, page, browser_n
         :id: d0efda45-4e8e-4703-b9c0-ab53249dafc3
         :setup: Standalone instance
         :steps:
-             1. Click on Database tab and check if ID List Scan Limit label is visible.
+             1. Click on Database tab, click on Limits tab and check if ID List Scan Limit label is visible.
              2. Click on Database Cache tab and check if Automatic Cache Tuning checkbox is visible.
              3. Click on Import Cache tab and check if Automatic Import Cache Tuning checkbox is visible.
              4. Click on NDN Cache tab and check if Normalized DN Cache Max Size label is visible.
@@ -69,31 +69,52 @@ def test_global_database_configuration_availability(topology_st, page, browser_n
     setup_login(page)
     time.sleep(1)
     frame = check_frame_assignment(page, browser_name)
-
-    log.info('Check if element on Limits tab is loaded.')
-    frame.get_by_role('tab', name='Database', exact=True).click()
-    frame.get_by_text('ID List Scan Limit', exact=True).wait_for()
-    assert frame.get_by_text('ID List Scan Limit', exact=True).is_visible()
-
-    log.info('Click on Database Cache tab and check if element is loaded')
-    frame.get_by_role('tab', name='Database Cache', exact=True).click()
-    assert frame.locator('#db_cache_auto').is_visible()
-
-    log.info('Click on Import Cache tab and check if element is loaded')
-    frame.get_by_role('tab', name='Import Cache', exact=True).click()
-    assert frame.locator('#import_cache_auto').is_visible()
-
-    log.info('Click on NDN Cache tab and check if element is loaded')
-    frame.get_by_role('tab', name='NDN Cache', exact=True).click()
-    assert frame.get_by_text('Normalized DN Cache Max Size').is_visible()
-
-    log.info('Click on Database Locks tab and check if element is loaded')
-    frame.get_by_role('tab', name='Database Locks', exact=True).click()
-    assert frame.locator('#dblocksMonitoring').is_visible()
-
-    log.info('Click on Advanced Settings tab and check if element is loaded')
-    frame.get_by_role('tab', name='Advanced Settings', exact=True).click()
-    assert frame.locator('#txnlogdir').is_visible()
+    instance = topology_st.standalone
+
+    if instance.get_db_lib() is 'mdb':
+        log.info('Check if element on Limits tab is loaded.')
+        frame.get_by_role('tab', name='Database', exact=True).click()
+
+        frame.get_by_role('tab', name='Database Size', exact=True).click()
+        frame.get_by_text('Database Maximum Size', exact=True).wait_for()
+        assert frame.get_by_text('Database Maximum Size', exact=True).is_visible()
+
+        frame.get_by_role('tab', name='Limits', exact=True).click()
+        frame.get_by_text('ID List Scan Limit', exact=True).wait_for()
+        assert frame.get_by_text('ID List Scan Limit', exact=True).is_visible()
+
+        log.info('Click on NDN Cache tab and check if element is loaded')
+        frame.get_by_role('tab', name='NDN Cache', exact=True).click()
+        assert frame.get_by_text('Normalized DN Cache Max Size').is_visible()
+
+        log.info('Click on Advanced Settings tab and check if element is loaded')
+        frame.get_by_role('tab', name='Advanced Settings', exact=True).click()
+        assert frame.locator('#dbhomedir').is_visible()
+    elif instance.get_db_lib() is 'bdb':
+        log.info('Check if element on Limits tab is loaded.')
+        frame.get_by_role('tab', name='Database', exact=True).click()
+        frame.get_by_text('ID List Scan Limit', exact=True).wait_for()
+        assert frame.get_by_text('ID List Scan Limit', exact=True).is_visible()
+
+        log.info('Click on Database Cache tab and check if element is loaded')
+        frame.get_by_role('tab', name='Database Cache', exact=True).click()
+        assert frame.locator('#db_cache_auto').is_visible()
+
+        log.info('Click on Import Cache tab and check if element is loaded')
+        frame.get_by_role('tab', name='Import Cache', exact=True).click()
+        assert frame.locator('#import_cache_auto').is_visible()
+
+        log.info('Click on NDN Cache tab and check if element is loaded')
+        frame.get_by_role('tab', name='NDN Cache', exact=True).click()
+        assert frame.get_by_text('Normalized DN Cache Max Size').is_visible()
+
+        log.info('Click on Database Locks tab and check if element is loaded')
+        frame.get_by_role('tab', name='Database Locks', exact=True).click()
+        assert frame.locator('#dblocksMonitoring').is_visible()
+
+        log.info('Click on Advanced Settings tab and check if element is loaded')
+        frame.get_by_role('tab', name='Advanced Settings', exact=True).click()
+        assert frame.locator('#txnlogdir').is_visible()
 
 
 def test_chaining_configuration_availability(topology_st, page, browser_name):
diff --git a/dirsrvtests/tests/suites/webui/ldap_browser/ldap_browser_test.py b/dirsrvtests/tests/suites/webui/ldap_browser/ldap_browser_test.py
index 265cc9489..0aab5c7c5 100644
--- a/dirsrvtests/tests/suites/webui/ldap_browser/ldap_browser_test.py
+++ b/dirsrvtests/tests/suites/webui/ldap_browser/ldap_browser_test.py
@@ -181,6 +181,7 @@ def test_create_and_delete_group(topology_st, page, browser_name):
     assert frame.get_by_role("button").filter(has_text=f"cn={test_data['group_name']}").count() == 0
 
 
+@pytest.mark.xfail(reason="DS6558")
 def test_create_and_delete_organizational_unit(topology_st, page, browser_name):
     """ Test to create and delete organizational unit
 
diff --git a/dirsrvtests/tests/suites/webui/monitoring/monitoring_test.py b/dirsrvtests/tests/suites/webui/monitoring/monitoring_test.py
index 40de3baeb..6a1556c71 100644
--- a/dirsrvtests/tests/suites/webui/monitoring/monitoring_test.py
+++ b/dirsrvtests/tests/suites/webui/monitoring/monitoring_test.py
@@ -132,6 +132,7 @@ def test_replication_visibility(topology_st, page, browser_name):
     assert frame.locator('#replication-suffix-dc\\=example\\,dc\\=com').is_visible()
 
 
+@pytest.mark.xfail(reason="DS6557")
 def test_database_visibility(topology_st, page, browser_name):
     """ Test Database monitoring visibility
 
@@ -247,7 +248,7 @@ def test_create_credential_and_alias(topology_st, page, browser_name):
 
     log.info('Click on Monitoring tab, click on replication button, create new credential and check if it is created')
     frame.get_by_role('tab', name='Monitoring', exact=True).click()
-    frame.locator('#replication-monitor').click()
+    frame.locator('#sync-report').click()
     frame.locator('#pf-tab-1-prepare-new-report').click()
     frame.get_by_role('button', name='Add Credentials').click()
     frame.locator('#credsHostname').fill('credential.test')
-- 
2.48.1

